File: happykopiApp/src/app/core/dtos/order/new-order-request.dto.ts
export interface NewOrderRequestDto {
  userId: number;
  totalAmount: number;
  status: string;
  paymentType: string;
  amountPaid: number;
  change: number;
  referenceNumber?: string | null;
  orderItems: NewOrderItemDto[];
}
 
export interface NewOrderItemDto {
  productVariantId: number;
  quantity: number;
  price: number;
  subtotal: number;
  modifiers: NewOrderModifierDto[];
}
 
export interface NewOrderModifierDto {
  modifierId: number;
  quantity: number;
  price: number;
  subtotal: number;
}

File: happykopiAPI/happykopiAPI/DTOs/Order/Ingoing Data/NewOrderRequestDTO.cs
namespace happykopiAPI.DTOs.Order.Ingoing_Data
{
    /// <summary>
    /// Main request DTO for creating a new order
    /// </summary>
    public class NewOrderRequestDto
    {
        public int UserId { get; set; }
        public decimal TotalAmount { get; set; }
        public string Status { get; set; } = "Completed"; // Default: Completed
        public string PaymentType { get; set; } // Cash, GCash, Card, etc.
        public decimal AmountPaid { get; set; }
        public decimal Change { get; set; }
        public string? ReferenceNumber { get; set; } // Optional for Cash
        public List<NewOrderItemDto> OrderItems { get; set; } = new();
    }
}